COMMANDLINE.md

```diff
@@ -0,0 +1,87 @@
+# Command-line Interface
+
+## Commands
+
+```
+help                    Display commands and usage.
+
+generateSolutions       Generate sudoku solution(s).
+  --amount {n >= 1}     Specify an amount to generate.
+  --normalize           Arrange digits such that the top row of the board
+                        reads 1 through 9 sequentially.
+  --threads             Use all available system threads.
+  --threads {n >= 1}    Use a given number of system threads.
+                        Not recommended to use more than the system maximum.
+  --pretty              Output grids in a more readable form.
+
+generatePuzzles         Generate sudoku puzzle(s).
+  --amount {n >= 1}     Specify an amount to generate.
+  --normalize           Arrange digits such that the top row of the solution
+                        reads 1 through 9 sequentially.
+  --threads             Use all available system threads.
+  --threads {n >= 1}    Use a given number of system threads.
+                        Not recommended to use more than the system maximum.
+  --clues {17 - 81}     Generate a puzzle with a given number of clues.
+                        Less than 20 is not recommended due to the
+                        processing power required.
+                        Default 31.
+  --difficulty {1 - 3}  Specify the difficulty of the generate puzzle.
+    1: Easy             Solvable by finding naked and hidden singles.
+    2: Moderate         Solvable by finding {Not yet defined}
+    3: Hard             Solvable with more advanced techniques than above.
```

Comment:
Fine as placeholder. #3 will need to clarify this line it's decided which techniques to implement